src/api/github/checkRun.ts

```diff
@@ -0,0 +1,120 @@
+import { Octokit } from '@octokit/rest';
+import { Endpoints } from '@octokit/types';
+
+type Output = NonNullable<
+  Endpoints['POST /repos/{owner}/{repo}/check-runs']['parameters']['output']
+>;
+
+export type Annotation = NonNullable<Output['annotations']>[number];
+
+const GITHUB_MAX_ANNOTATIONS = 50;
+
+const isGitHubAnnotationsEnabled = (): boolean =>
+  Boolean(
+    process.env.BUILDKITE_BUILD_NUMBER &&
+      process.env.BUILDKITE_COMMIT &&
+      process.env.BUILDKITE_REPO &&
+      process.env.GITHUB_API_TOKEN,
+  );
+
+// Pulls out the GitHub Owner + Repo String from repo urls eg.
+// git@github.com:seek-oss/skuba.git
+// https://github.com/seek-oss/skuba.git
+// Pulls out `seek-oss` as owner and `skuba` as repo
+const ownerRepoRegex = new RegExp(/github.com(?::|\/)(.*)\/(.*).git/);
```

Comment:
```suggestion
/**
 * Matches the owner and repository names in a GitHub repository URL.
 *
 * For example, given the following input strings:
 *
 * ```console
 * git@github.com:seek-oss/skuba.git
 * https://github.com/seek-oss/skuba.git
 * ```
 *
 * This pattern will produce the following matches:
 *
 * 1. seek-oss
 * 2. skuba
*/
const ownerRepoRegex = new RegExp(/github.com(?::|\/)(.+)\/(.+).git$/);
```